api/cpp/platform.rs

```diff
@@ -400,6 +402,87 @@ mod software_renderer {
         i_slint_core::graphics::euclid::rect(orig.x, orig.y, size.width as i32, size.height as i32)
     }
 
+    #[no_mangle]
+    pub unsafe extern "C" fn slint_software_renderer_render_by_line_rgb565(
+        r: SoftwareRendererOpaque,
+        process_line_fn: extern "C" fn(
+            *mut core::ffi::c_void,
+            usize,
+            usize,
+            usize,
+            extern "C" fn(*const core::ffi::c_void, *mut u16, usize),
+            *const core::ffi::c_void,
+        ),
+        user_data: *mut core::ffi::c_void,
+    ) -> IntRect {
+        struct Rgb565Processor {
```

Comment:
Could be nice if this was a `struct LineProcessor<TargetPixel>` so that it can be re-used in a slint_software_renderer_render_by_line_rgb888